src/main/php/org/bovigo/vfs/content/SeekableFileContent.php

```diff
@@ -52,24 +52,30 @@ public function read($count)
      */
     public function seek($offset, $whence)
     {
+        $old_offset = $this->offset;
         switch ($whence) {
             case SEEK_CUR:
                 $this->offset += $offset;
-                return true;
+                break;
 
             case SEEK_END:
                 $this->offset = $this->size() + $offset;
-                return true;
+                break;
 
             case SEEK_SET:
                 $this->offset = $offset;
-                return true;
+                break;
 
             default:
                 return false;
         }
+        
+        if ($this->offset<0) {
+            $this->offset = $old_offset;
+            return false;
+        }
 
```

Comment:
I believe it should be the other way around. Instead of changing the internal offset value and reset it if it's not valid, the new value should be calculated, then checked if it is valid, and only when it is valid the internal offset value should be changed to the new value. This way data corruption can be prevented in case something stupid happens, and it also makes it more failure proof against future changes.

Also, let's stick to camelCase for variable names as all other variable names are already camelCase.